lib/clientv2/networkpolicy.go

```diff
@@ -41,8 +42,16 @@ type networkPolicies struct {
 // representation of the NetworkPolicy, and an error, if there is any.
 func (r networkPolicies) Create(ctx context.Context, res *apiv2.NetworkPolicy, opts options.SetOptions) (*apiv2.NetworkPolicy, error) {
 	defaultPolicyTypesField(&res.Spec)
+
+	// Properly prefix the name
+	name := res.GetObjectMeta().GetName()
+	defer resetPolicyName(res, name)
```

Comment:
You don't need this defer anymore.

I think things would feel a lot more symmetric here if you had a convertPolicyNameForStorage and convertPolicyNameFromStorage  or something similar.   You've got one method in one direction and on in another.

That way I think this code would look something like:

`res.GetObjectMeta().SetName(convertPolicyNameForStorage(res.GetObjectMeta().GetName()))`

and just before you return the result you do the same with the opposite conversion function.  That feels like it hides less away.

Obviously for the list you'd need to go through each Item and convert those.

Reply:
I'm not suggesting mine is the right way, but it feels cleaner to me than what we have at the moment since we have a number of different methods used at different times performing subtly different things.

Reply:
I agree that it would be easier to read with symmetric name conversion functions.  Thinking about it, the only reason I can see keeping the way its written now is if for some reason the return value from the client for some reason does not follow the `<prefix>.<name>` format. How valid of a case would it be that data is injected into etcd (I'm not sure if you can do this to KDD) that would not follow our internal prefixing structure and is named something like `<something>.<random>`?

Reply:
Actually, that's a silly concern because if we convert correctly, then this wouldn't find the strange name that they might manually inject.